### PR TITLE
Add cairo1 invoke test

### PR DESCRIPTION
**Pull request by karol-bisztyga**

### CODE REVIEW

**Comment by karol-bisztyga** [comment]
CI fails bcoz https://github.com/software-mansion-labs/cairo/pull/21 ain't merged

**Review by MaksymilianDemitraszek — CHANGES_REQUESTED** [review]

File: tests/integration/cairo1_compiler_bindings/library_functions_test.py [hunk 1]
```diff
@@ -69,3 +69,7 @@ def test_declare(mocker: MockerFixture, datadir: Path):
 
 def test_start_prank(mocker: MockerFixture, datadir: Path):
     check_library_function(mocker, "start_prank", datadir / "start_prank_test.cairo")
+
+
+def test_invoke(mocker: MockerFixture, datadir: Path):
```
[the remark this answers is not in the document]
Comment (karol-bisztyga) [reply tests/integration/cairo1_compiler_bindings/library_functions_test.py]: [add args validator](https://github.com/software-mansion/protostar/pull/1527/commits/3495dd39ed06d79a03da7eca4de96449a15ac2ed)

File: tests/integration/cairo1_compiler_bindings/library_functions_test.py [hunk 2]
```diff
@@ -59,17 +72,37 @@ def check_library_function(
...
-    check_library_function(mocker, "start_prank", datadir / "start_prank_test.cairo")
 
+def test_warp(datadir: Path):
+    check_library_function("warp", datadir / "warp_test.cairo")
 
-def test_warp(mocker: MockerFixture, datadir: Path):
-    check_library_function(mocker, "warp", datadir / "warp_test.cairo")
+
+def test_invoke(datadir: Path):
+    def args_validator(memory: MemoryDict, *args: Any, **kwargs: Any):
```
Comment (MaksymilianDemitraszek) [inline tests/integration/cairo1_compiler_bindings/library_functions_test.py:92]: I agree that this seems like a good usecase for mocks

File: tests/integration/cairo1_compiler_bindings/library_functions_test/invoke_test.cairo [hunk 3]
```diff
@@ -0,0 +1,22 @@
+use array::ArrayTrait;
+
+#[test]
+fn test_invoke() {
+   let mut arr = ArrayTrait::new();
+   arr.append(101);
```
Comment (MaksymilianDemitraszek) [inline tests/integration/cairo1_compiler_bindings/library_functions_test/invoke_test.cairo:6]: Can we create a second test with an empty array?

**Review by MaksymilianDemitraszek — CHANGES_REQUESTED** [review]

File: tests/integration/cairo1_compiler_bindings/library_functions_test.py [hunk 4]
```diff
@@ -59,17 +72,37 @@ def check_library_function(
...
+        assert contract_address == 123
+        expected_calldata = [101, 202, 303, 405, 508, 613, 721]
+        actual_calldata = []
+        calldata_start = memory.data[kwargs["calldata_start"][0]]
+        calldata_end = memory.data[kwargs["calldata_end"][0]]
+        iterator = calldata_start
+        while iterator != calldata_end:
+            actual_calldata.append(memory.data[iterator])
+            iterator = iterator + 1
+        assert actual_calldata == expected_calldata
```
[the remark this answers is not in the document]
Comment (MaksymilianDemitraszek) [reply tests/integration/cairo1_compiler_bindings/library_functions_test.py:104]: I think it includes logic of extracting it from memory, so mock is not a good solution here

**Review by MaksymilianDemitraszek — APPROVED** [review]

File: tests/integration/cairo1_compiler_bindings/library_functions_test/invoke_test.cairo (hunk 3 above)
Comment (karol-bisztyga) [reply tests/integration/cairo1_compiler_bindings/library_functions_test/invoke_test.cairo:6]: [62fa1d4](https://github.com/software-mansion/protostar/pull/1527/commits/62fa1d411ae0d6c53089b78c8bee7a61e654a8f5)

File: tests/integration/cairo1_compiler_bindings/library_functions_test.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (karol-bisztyga) [reply tests/integration/cairo1_compiler_bindings/library_functions_test.py]: @Arcticae are we good here? Can we resolve this?

[the remark this answers is not in the document]
Comment (karol-bisztyga) [reply tests/integration/cairo1_compiler_bindings/library_functions_test.py]: Discussed offline, resolving

**Pull request merged**